src/powershell/Public/RegisterFinOpsHubProviders.ps1

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+<#
+.SYNOPSIS
+    Registers Azure resource providers required for FinOps hub.
+
+.PARAMETER WhatIf
+    Optional. Shows what would happen if the command runs without actually running it.
+
+.EXAMPLE
+    Register-FinOpsHubProviders -WhatIf
+
+    Shows what would happen if the command runs without actually running it.
+
+.NOTES
+    This command registers the following Azure resource providers required for FinOps hub:
+    - Microsoft.EventGrid
+    - Microsoft.CostManagementExports
+#>
+function Register-FinOpsHubProviders {
+    [CmdletBinding()]
+    param (
+        [Switch]$WhatIf
```

Comment:
PowerShell has a built-in mechanism to provide this functionality. It would be a CmdletBinding attribute: [CmdletBinding(SupportsShouldProcess)]. https://learn.microsoft.com/en-us/powershell/scripting/learn/deep-dives/everything-about-shouldprocess?view=powershell-7.3#supportsshouldprocess

Reply:
Changed it.